### ISSUE TITLE
Replace `autoClosableRetainsCloseButton` alert screenshot test with e2e

### ISSUE THREAD

**Issue by Elijbet** [Esri/calcite-design-system#9122]
**monday.com sync:** #18048702795

### Description

Dropped `autoClosableRetainsCloseButton_TestOnly` in PR #9118 to be replaced with an e2e test asserting close button is retained throughout the length of close delay and closes when prompted. 

### Proposed Advantages

e2e test is more appropriate for this functionality

### Which Component

alert

### Calcite package

- [ ] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

**Comment by jcfranco** [comment]
@Elijbet Can you link to the PR that removed the `autoClosableRetainsCloseButton_TestOnly` test?

**Comment by Elijbet** [comment]
> @Elijbet Can you link to the PR that removed the `autoClosableRetainsCloseButton_TestOnly` test?

Yes. It's [test(alert, text-area): improve layout for screenshot diffs](https://github.com/Esri/calcite-design-system/pull/9118#top)
#9118

**Comment by DitwanP** [comment]
👌✨ Verified locally
